src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,75 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { User } from '../../types/User';
+import { actions } from '../../features/currentTodo';
+import { getUser } from '../../api';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [user, setUser] = useState<User | null>(null);
+
+  const resetCurrentTodo = () => {
+    dispatch(actions.removeTodo());
+  };
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      if (currentTodo) {
+        const userFromServer = await getUser(currentTodo.userId);
+
+        setUser(userFromServer);
+      }
+    };
+
+    fetchUser();
+  }, []);
```

Comment:
The useEffect dependency array is empty, causing the fetchUser function to only run once. Consider adding 'currentTodo' as a dependency to refetch the user when the currentTodo changes.